providers/go-feature-flag/src/main/java/dev/openfeature/contrib/providers/gofeatureflag/GoFeatureFlagProvider.java

```diff
@@ -322,4 +424,45 @@ private Structure mapToStructure(Map<String, Object> map) {
                         .filter(e -> e.getValue() != null)
                         .collect(Collectors.toMap(Map.Entry::getKey, e -> objectToValue(e.getValue()))));
     }
+
+    private void publishEventsQuietly(List<Event> eventsList) {
+        try {
+            publishEvents(eventsList);
+        } catch (Exception e) {
+            log.error("Error publishing events", e);
+        }
+    }
+
+    private void publishEvents(List<Event> eventsList) throws Exception {
+        Events events = new Events(eventsList);
+        HttpUrl url = this.parsedEndpoint.newBuilder()
+            .addEncodedPathSegment("v1")
+            .addEncodedPathSegment("data")
+            .addEncodedPathSegment("collector")
+            .build();
+
+        Request.Builder reqBuilder = new Request.Builder()
+            .url(url)
+            .addHeader("Content-Type", "application/json")
+            .post(RequestBody.create(
+                requestMapper.writeValueAsBytes(events),
+                MediaType.get("application/json; charset=utf-8")));
+
+        if (this.apiKey != null && !"".equals(this.apiKey)) {
+            reqBuilder.addHeader("Authorization", "Bearer " + this.apiKey);
+        }
+
+        try (Response response = this.httpClient.newCall(reqBuilder.build()).execute()) {
+            if (response.code() == HTTP_UNAUTHORIZED) {
+                throw new GeneralError("Unauthorized");
```

Comment:
I am not sure we should throw in case the data collection is not working.
What I would prefer is that we are keeping the data in memory and we retry. A log can still be welcome for this.

What do you think about it?

Reply:
The caller is _publishEventsQuietly_, which is catching and logging the exception without failing the flow.
Regarding retry, would like some details:
1. The data is used only for statistics at the server side ?
2. What is the retry logic in _go-sdk_ for publishing these events ?
3. If retry is used, it depends on the result, for example if we retry on invalid bad request, it is expected to fail again.